tools/utils/test/utils.ts

```diff
@@ -95,4 +101,25 @@ describe('Utils', async () => {
       assert(true)
     }
   })
+
+  it('Compresses and decompresses an order', async () => {
+    const { v, r, s } = await createSwapSignature(
+      unsignedOrder,
+      wallet.privateKey,
+      ADDRESS_ZERO,
+      1
+    )
+    const signerWallet = getSignerFromSwapSignature(
+      unsignedOrder,
+      ADDRESS_ZERO,
+      1,
+      v,
+      r,
+      s
+    )
+    const compressed = compressOrder({ ...unsignedOrder, v, r, s })
+    const signedOrder = decompressOrder(compressed)
+    expect(isValidOrder(signedOrder)).to.equal(true)
+    expect(signerWallet.toLowerCase()).to.equal(wallet.address.toLowerCase())
```

Comment:
Could do:

```typescript
expect(signedOrder).to.deep.equal({ ...unsignedOrder, v, r, s})
```

Reply:
This would fail because `signedOrder` (type `Order`) does not include `protocolFee` or `senderWallet` as they are respectively compared to the `protocolFee` on the Swap contract and `msg.sender` during transaction execution.

However, compressed order strings could include these additional params (and others) if necessary for UX, even if they're not necessary to actually take the order.